cypress/e2e/read-receipts/read-receipts.spec.ts

```diff
@@ -354,4 +357,711 @@ describe("Read receipts", () => {
+
+                // When I read the main timeline
+                goTo(room2);
+
+                // Then room does appear unread
+                assertUnread(room2, 2);
+
+                // Until we open the thread
+                openThread("Msg1");
+                assertReadThread("Msg1");
+                assertRead(room2);
+            });
+            it.skip("Reading an older thread message (via permalink) leaves the thread unread", () => {});
+            it("Reading only one thread's message does not make the room read", () => {
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", threadedOff("Msg1", "Resp1"), "Msg2", threadedOff("Msg2", "Resp2")]);
+                assertUnread(room2, 4);
+                goTo(room2);
+                assertUnread(room2, 4);
+
+                openThread("Msg1");
+                assertUnread(room2, 1);
+            });
+            it("Reading only one thread's message make that thread read but not others", () => {
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", "Msg2", threadedOff("Msg1", "Resp1"), threadedOff("Msg2", "Resp2")]);
+                assertUnread(room2, 4); // (Sanity)
+
+                // When I read the main timeline
+                goTo(room2);
+
+                assertUnread(room2, 2);
+                assertUnreadThread("Msg1");
+                assertUnreadThread("Msg2");
+
+                openThread("Msg1");
+                assertReadThread("Msg1");
+                assertUnreadThread("Msg2");
+            });
+            it("Reading the main timeline does not mark a thread message as read", () => {
+                // Given a thread exists
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", threadedOff("Msg1", "Resp1"), threadedOff("Msg1", "Resp2")]);
+                assertUnread(room2, 3); // (Sanity)
+
+                // When I read the main timeline
+                goTo(room2);
+
+                assertUnread(room2, 2);
+                // Then thread does appear unread
+                assertUnreadThread("Msg1");
+            });
+            // XXX: this failure seems legit
+            it.skip("Marking a room with unread threads as read makes it read", () => {
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", threadedOff("Msg1", "Resp1"), threadedOff("Msg1", "Resp2")]);
+                assertUnread(room2, 3); // (Sanity)
+
+                markAsRead(room2);
+                assertRead(room2);
+            });
+            it("Sending a new thread message after marking as read makes it unread", () => {
+                // Given a thread exists
+                goTo(room1);
+                receiveMessages(room2, ["Msg1", threadedOff("Msg1", "Resp1"), threadedOff("Msg1", "Resp2")]);
+
+                // When I read the main timeline
+                goTo(room2);
+
+                // And the thread
+                openThread("Msg1");
```

Comment:
My intent here was to use mark as read, but we need to figure out the question about the previous test before we can figure this on out.

Reply:
I did copy paste a lot, feel free to update

Reply:
Updated